src/cargo/ops/cargo_add/mod.rs

```diff
@@ -537,19 +538,67 @@ fn get_latest_dependency(
+            ) -> Option<&'a Summary> {
+                possibilities.max_by_key(|s| {
+                    // Fallback to a pre-release if no official release is available by sorting them
+                    // as less.
                     let stable = s.version().pre.is_empty();
                     (stable, s.version())
                 })
-                .ok_or_else(|| {
-                    anyhow::format_err!(
-                        "the crate `{dependency}` could not be found in registry index."
-                    )
-                })?;
+            }
+
+            let mut latest = find_latest(possibilities.iter()).ok_or_else(|| {
+                anyhow::format_err!(
+                    "the crate `{dependency}` could not be found in registry index."
+                )
+            })?;
+
+            if config.cli_unstable().msrv_policy {
+                if let Some(rust_version) = manifest
+                    .data
+                    .as_table()
+                    .get("package")
+                    .and_then(|p| p.get("rust-version"))
+                    .and_then(|v| v.as_str())
+                    .and_then(|v| semver::VersionReq::parse(v).ok())
+                {
+                    // Find the latest version of the dep which has a compatible MSRV. Candidates
+                    // without a rust-version field are treated as compatible.
+                    let latest_msrv = find_latest(possibilities.iter().filter(|s| {
+                        s.rust_version()
+                            .map(|v| v.as_str())
+                            .map(|v| semver::VersionReq::parse(v).unwrap())
+                            .map(|v| {
+                                // Rust versions need at least a major and a minor value, and cannot
+                                // contain semver operators or pre-release identifiers, so the
+                                // comparison is simple.
+                                let required = &rust_version.comparators[0];
+                                let other = &v.comparators[0];
+                                (required.major, required.minor.unwrap(), required.patch)
+                                    >= (other.major, other.minor.unwrap(), other.patch)
+                            })
+                            .unwrap_or(true)
+                    }))
+                    .ok_or_else(|| {
+                        anyhow::format_err!(
+                            "could not find version of crate `{dependency}` that satisfies the \
+                             minimum supported rust version"
+                        )
+                    })?;
+
+                    if latest_msrv.version() < latest.version() {
+                        config.shell().warn(format!(
+                            "selecting older version of `{dependency}` to satisfy the minimum \
+                             supported rust version"
```

Comment:
Similarly, we should report what the latest version is, what its MSRV is, and the `--ignore-rust-version` flag.

This can also help clear up the message a bit: "ignoring {name} {latest} due to rust-version {dep_msrv}.  To get the latest, either update rust-version from {pkg_msrv} or use `--ignore-rust-version`"

(it doesn't have to be literally that, just an example"